File: samples/dart/01table/table.dart
import "dart:io";
import 'package:sprintf/sprintf.dart';


void main(List<String> arguments) {

    const limit = 5;

    stdout.write("   ");

    for (int i = 1; i <= limit; i++) {
        stdout.write(sprintf("%-3d", [i]));
    }
    print("");

    for (int a = 1; a <= limit; a++) {
        stdout.write(sprintf("%-3d", [a]));
        for (int b = 1; b <= limit; b++) {
            stdout.write(sprintf("%-3d", [a*b]));
        }
        print("");
    }
}

File: samples/js/01table/table.js
let __left_cache = {};
function leftPad(str, n, { padCharacter=" " }={}) {
    padCharacter = String(padCharacter);
    if (padCharacter.length != 1) {
        throw "Invalid padCharacter. Should be one character";
    }
    str = String(str);
    if (str.length < n) {
        while (str.length < n) {
            str = padCharacter + str;
        }
    }
    return str;
}



if (require.main == module) {
    const LIMIT = 5;
    for (let a = 0; a <= LIMIT; a++) {
        if (a == 0) {
            process.stdout.write(leftPad("", 3));
        } else {
            process.stdout.write(leftPad(a, 3));
        }
        for (let b = 0; b <= LIMIT; b++) {
            if (a == 0) {
                process.stdout.write(leftPad(b, 3));
            } else {
                process.stdout.write(leftPad(a*b, 3));
            }
        }
        console.log();
    }
}
